### PR TITLE
docs: update testing document to include karma config

### PR DESCRIPTION
**Pull request by alan-agius4**
In version 15, the Angular CLI will no longer generate a `karma.conf.js` by default.

This commit updates the testing document to include an example of how to include a custom `karma.conf.js`.  Also, we update several parts of the documentation which were outdated.

### CODE REVIEW

File: aio/content/guide/testing.md [hunk 1]
```diff
@@ -75,28 +73,89 @@ The tests run again, the browser refreshes, and the new test results appear.
...
+        singleRun: false,
+        restartOnFileChange: true
+      });
+    };
 
-Fine-tune many options by editing the `karma.conf.js` in the root folder of the project and
-the `test.ts` files in the `src/` folder.
+    </code-example>
+
+1. In `angular.json`, configure the Karma builder to use the created configuration file.
```
Comment (dgp1130) [inline aio/content/guide/testing.md:128]: This is an existing builder right? We should call out the new `karmaConfig` option in particular to draw attention to that as the change. Do we happen to have a `diff` type which gets rendered in AIO? That would be good to use here if it's supported.

File: aio/content/guide/testing.md [hunk 2]
```diff
@@ -75,28 +73,89 @@ The tests run again, the browser refreshes, and the new test results appear.
...
+      "karmaConfig": "karma.conf.js",
+      "polyfills": ["zone.js", "zone.js/testing"],
+      "tsConfig": "src/tsconfig.spec.json",
+      "styles": ["src/styles.css"]
+    }
+  }
 
-The `karma.conf.js` file is a partial Karma configuration file.
-The CLI constructs the full runtime configuration in memory, based on application structure specified in the `angular.json` file, supplemented by `karma.conf.js`.
+  </code-example>
```
Comment (dgp1130) [inline aio/content/guide/testing.md:142]: We do run the risk of these code samples falling out of date, but we shouldn't update them too often (majors/minors at most) so I'm not too worried about it. Something to keep in mind if we do end up changing these though.

**Review by dgp1130 — APPROVED** [review]

File: aio/content/guide/testing.md (hunk 1 above)
Comment (alan-agius4) [reply aio/content/guide/testing.md:128]: diff does not appear to be supported. I called out the karmaConfig option.

File: aio/content/guide/testing.md [hunk 3]
```diff
@@ -197,202 +197,17 @@ next to their corresponding helper files.
...
-    </code-example>
+Continuous integration \(CI\) servers let you set up your project repository so that your tests run on every commit and pull request.
 
-Now, run the following command to use the `--no-sandbox` flag:
+To test your Angular CLI application in Continuous integration \(CI\) run the following command:
 
 <code-example format="shell" language="shell">
 
-ng test --no-watch --no-progress --browsers=ChromeHeadlessCI
+ng test --no-watch --no-progress
```
Comment (dgp1130) [inline aio/content/guide/testing.md:210]: Don't we do this automatically when the `CI` variable is set? Do we need to explicitly specify these options, or is this just for CI providers which don't set the `CI` environment variable?

**Review by dgp1130 — APPROVED** [review]

File: aio/content/guide/testing.md (hunk 3 above)
Comment (dgp1130) [reply aio/content/guide/testing.md:210]: Maybe this is something we should revisit then. It's always bugged me that `ng test` opened a visible browser and ran in watch mode. I was under the impression the `CI` variable at least make it work reasonably in this context. We can discuss this more and see if it's something we want to change.

**Review by dgp1130 — APPROVED** [review]

**Review by jessicajaniuk — APPROVED** [review]
LGTM

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 665af467d88ad7bf5eb457617a2ce9f6963d22dd.

**Pull request closed without merging**